app/scripts/controllers/transactions/index.js

```diff
@@ -781,18 +776,30 @@ export default class TransactionController extends EventEmitter {
    *  fragment for
    * @param {valueOf<TransactionMetaMetricsEvent>} event - event type to create
    * @param {string} actionId - actionId passed from UI
+   * @param {boolean} isFragmentExists - whether fragment exist or not
+   * @param {object} updateParams - Update payload sent from UI
    */
-  async createTransactionEventFragment(transactionId, event, actionId) {
+  async createOrUpdateTransactionEventFragment(
```

Comment:
The end result of this ticket should be that we have no API in the transaction controller concerning metrics as that is to be handled in the client itself.

Can we remove this method from the public API and communicate all the required data through events?